public/app/plugins/datasource/prometheus/datasource.test.ts

```diff
@@ -1619,17 +1619,43 @@ describe('PrometheusDatasource', () => {
   });
 
   describe('The __rate_interval variable', () => {
+    const target = { expr: 'rate(process_cpu_seconds_total[$__rate_interval])', refId: 'A' };
+
+    beforeEach(() => {
+      (templateSrv.replace as any).mockClear();
+    });
+
     it('should be 4 times the scrape interval if interval + scrape interval is lower', () => {
-      const { __rate_interval } = ds.getRateIntervalScopedVariable(23, 23);
-      expect(__rate_interval.value).toBe('60s');
+      ds.createQuery(target, { interval: '15s' } as any, 0, 300);
+      expect((templateSrv.replace as any).mock.calls[1][1]['__rate_interval'].value).toBe('60s');
     });
     it('should be interval + scrape interval if 4 times the scrape interval is lower', () => {
-      const { __rate_interval } = ds.getRateIntervalScopedVariable(56, 56);
-      expect(__rate_interval.value).toBe('71s');
-    });
-    it('should fall back to 60s if interval is 0', () => {
-      const { __rate_interval } = ds.getRateIntervalScopedVariable(0, 0);
-      expect(__rate_interval.value).toBe('60s');
+      ds.createQuery(target, { interval: '5m' } as any, 0, 10080);
+      expect((templateSrv.replace as any).mock.calls[1][1]['__rate_interval'].value).toBe('315s');
+    });
+    it('should fall back to 60s if interval and scrapeInterval is 0', () => {
```

Comment:
This is actually not quite accurate now because the interval is not 0. 

I guess it should be: 'should fall back to a scrape interval of 15s if min step is set to 0, resulting in 4*15s = 60s'